### ISSUE TITLE
Uncaught Error: Cannot decorate a destroyed marker

### ISSUE THREAD

**Issue by abraham** [steelbrain/linter-ui-default#434]
Closed 5 tabs in quick succession and got this error.

Atom: 1.20.0 x64
Electron: 1.6.9
OS: Mac OS X 10.12.6
Thrown From: linter-ui-default package 1.6.8

Stack Trace

Uncaught Error: Cannot decorate a destroyed marker

```
At /Applications/Atom.app/Contents/Resources/app/src/decoration-manager.js:191

Error: Cannot decorate a destroyed marker
    at DecorationManager.decorateMarker (/Applications/Atom.app/Contents/Resources/app/src/decoration-manager.js:183:27)
    at TextEditor.module.exports.TextEditor.decorateMarker (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1811:43)
    at /packages/linter-ui-default/lib/tooltip/index.js:33:16)
    at Editor.updateTooltip (/packages/linter-ui-default/lib/editor/index.js:238:20)
    at /packages/linter-ui-default/lib/editor/index.js:194:12
    at later (/packages/linter/node_modules/sb-debounce/index.js:9:14)
```

Commands

```
     -7:23.1.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -7:22.7.0 core:backspace (input.hidden-input)
     -7:20.9.0 core:save (input.hidden-input)
     -7:04.2.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -7:04.0 core:backspace (input.hidden-input)
     -6:57.9.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -6:57.7.0 core:backspace (input.hidden-input)
  3x -6:56.8.0 core:save (input.hidden-input)
     -5:11.4.0 fuzzy-finder:toggle-file-finder (input.hidden-input)
     -5:11.2.0 core:paste (input.hidden-input)
     -5:10.7.0 core:confirm (input.hidden-input)
  2x -5:07.5.0 core:backspace (input.hidden-input)
     -5:05.3.0 editor:delete-to-beginning-of-line (input.hidden-input)
     -5:05.1.0 core:backspace (input.hidden-input)
     -5:04.8.0 core:save (input.hidden-input)
  5x -0:52.1.0 core:close (input.hidden-input)
```

Non-Core Packages

```
atom-beautify 0.30.5 
atom-bootstrap3 1.2.12 
atom-jshint 2.0.0 
atom-material-syntax 1.0.6 
atom-material-ui 2.0.4 
atom-polymer 0.13.0 
atom-ternjs 0.18.3 
atom-typescript 11.0.8 
autocomplete-ruby 0.2.8 
busy-signal 1.4.3 
file-icons 2.1.11 
git-time-machine 1.5.9 
highlight-selected 0.13.1 
intentions 1.1.5 
language-ampscript 0.5.0 
language-crontab 1.0.0 
language-docker 1.1.8 
language-gherkin 1.0.4 
language-graphql 0.9.0 
language-haml 0.25.1 
language-slim 0.6.0 
language-typescript-grammars-only 1.6.0 
linter 2.2.0 
linter-erb 1.1.0 
linter-eslint 8.2.1 
linter-flake8 2.2.1 
linter-htmlhint 1.3.3 
linter-js-yaml 1.2.7 
linter-jscs 4.1.3 
linter-php 1.4.0 
linter-phpmd 1.5.3 
linter-polymer 0.0.3 
linter-pycodestyle 2.1.3 
linter-rubocop 2.1.1 
linter-ruby 1.2.6 
linter-tslint 1.7.0 
linter-twig 0.4.2 
linter-ui-default 1.6.8 
material-monokai-syntax 1.0.1 
merge-conflicts 1.4.5 
monokai 0.24.0 
php-twig 4.0.0 
pigments 0.40.2 
polymer-atom 0.0.8 
polymer-ide 0.4.0 
polymer-snippets 3.2.1 
prettier-atom 0.38.0 
pretty-json 1.6.4 
rails-rspec 0.3.2 
rspec 0.4.0 
ruby-solargraph 0.1.0 
sort-lines 0.15.0 
Sublime-Style-Column-Selection 1.7.4 
travis-ci-status 1.3.1 
zentabs 0.8.8 
```

Similar to #224

**Comment by revelt** [comment]
Having the same problem. It surfaced very recently. Didn't have this before.

**Comment by jonpspri** [comment]
When I restarted, I happened to see that linter reinstalled `busy-signal`, which I'd removed in a fit of housekeeping.  I wonder if there's a correlation?

**Comment by Arcanemagus** [comment]
Was a provider still checking the files when you closed the tabs? It looks like there was a 5 second delay which _should_ have been plenty of time, but maybe not for a slow provider.

Trying to figure out how you triggered this.

@jonpspri That has nothing to do with this. (If you don't want that to be installed answer `Never` on the prompt for installing that dependency. You can manually edit the list following the instructions [here](https://github.com/steelbrain/package-deps/blob/e600c248d59c01d3dd7e1b7b984b7875e52f4095/src/helpers.js#L144).)

**Comment by greenais** [comment]
The same here but without closing tabs, just typing some code.

**Comment by revelt** [comment]
@greenais did you update to the latest? Are you on beta? I'm on latest beta and it stopped around two weeks ago. I don't get this error these days.

**Comment by greenais** [comment]
Yer, I did, autoupdate is on.

Atom 1.23.1
linter-ui-default 1.6.10

**Comment by revelt** [comment]
@greenais would you like to try the beta?

**Comment by greenais** [comment]
Why not? Now then?
Meanwhile there weren't those errors these days..

**Comment by skylize** [comment]
Just to add a tiny data point that might help point to resolving this at some time. I was just now receiving this error any time I focused a json file (which lints with `linter-jsonlint`, while still having no issues with js files being linted by `linter-eslint`. 

Problem went away for me after restarting Atom. No idea whether it was somehow _caused by_ `json-lint` or if it just persists in whatever file scope was active at the time the bug was triggered.

**Comment by skylize** [comment]
Definitely not tied to a specfic linter. Previously received this on json files, now receiving this now on js file. 

Once triggered, it seems to persists on the specific file(s) it originally occurred on, even after buffer is destroyed. After the error is thrown, if you close the tab that threw the error and reopen the same file, it will throw again each time on that file until Atom is restarted.

**Comment by steelbrain** [comment]
A potential fix just landed for this in v1.7.0
